### PR TITLE
Improve on multiconditionals/switch/select case

### PR DESCRIPTION
**Pull request by MichaelSt98**
Improve on multiconditionals/switch/select case for c-like-backends ...
this includes handling of RangeIndex as case value

### CODE REVIEW

**Review by reuterbal — COMMENTED** [review]
Neat addition! I left a question otherwise GTG!

File: loki/backend/cgen.py [hunk 1]
```diff
@@ -458,28 +458,83 @@ def visit_MultiConditional(self, o, **kwargs):
...
+                if not isinstance(val, sym.RangeIndex):
+                    sub_cases.append(self.visit(val, **kwargs))
+                else:
+                    assert (val.lower is None or isinstance(val.lower, sym.IntLiteral))\
+                            and isinstance(val.upper, sym.IntLiteral)
+                    lower = val.lower.value if val.lower is not None else 0
+                    sub_cases.extend([str(v) for v in list(range(lower, val.upper.value + 1))])
+            case = ()
+            for sub_case in sub_cases:
+                case += (self.format_line('case ', self.join_items(as_tuple(sub_case)), ':'),)
```
[the remark this answers is not in the document]
Comment (MichaelSt98) [reply loki/backend/cgen.py:532]: No particular reason ...

**Review by reuterbal — APPROVED** [review]
Thanks for clarifying. LGTM

**Review by mlange05 — APPROVED** [review]
Very neat indeed. Looks great, GTG. :shipit:

**Pull request merged**